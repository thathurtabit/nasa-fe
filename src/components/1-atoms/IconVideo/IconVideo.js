import React from 'react';
import PropTypes from 'prop-types';
import SVG from './IconVideo.styled';

const IconVideo = props => (
  <SVG width={50} height={50} viewBox="0 0 25 25" {...props}>
    <title>{props.title}</title>
    <path d="M12 2c5.514 0 10 4.486 10 10s-4.486 10-10 10S2 17.514 2 12 6.486 2 12 2zm0-2C5.373 0 0 5.373 0 12s5.373 12 12 12 12-5.373 12-12S18.627 0 12 0zM9 17V7l9 5.146L9 17z" />
  </SVG>
);

export default IconVideo;

IconVideo.propTypes = {
  title: PropTypes.string.isRequired,
};
