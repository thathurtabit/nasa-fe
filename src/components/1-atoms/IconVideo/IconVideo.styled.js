import styled from 'styled-components';

const SVG = styled.svg`
  fill: ${props => props.theme.colors.copy};
  position: relative;
`;

export default SVG;
