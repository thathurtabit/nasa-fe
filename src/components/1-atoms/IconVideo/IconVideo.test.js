import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import 'jest-styled-components';
import { shallowWithTheme } from '../../../utils/test_config/testHelpers';

import IconVideo from './IconVideo';

const title = 'test title';

describe('IconVideo', () => {
  it('should render correctly', () => {
    const element = shallowWithTheme(<IconVideo title={title} />);
    expect(shallowToJson(element)).toMatchSnapshot();
  });
});
