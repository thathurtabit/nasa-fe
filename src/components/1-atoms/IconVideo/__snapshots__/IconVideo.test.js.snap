// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconVideo should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
